packages/webvh/src/dids/WebvhDidRegistrar.ts

```diff
@@ -0,0 +1,239 @@
+import type {
+  AgentContext,
+  DidRegistrar,
+  DidCreateOptions,
+  DidCreateResult,
+  DidDeactivateResult,
+  DidDocument,
+  DidUpdateResult,
+  DidUpdateOptions,
+  DidDocumentRole,
+} from '@credo-ts/core'
+
+import { MultibaseEncoder, Buffer, DidRepository, DidRecord, isValidPrivateKey, KeyType } from '@credo-ts/core'
+import { createDID, updateDID } from 'didwebvh-ts'
+
+import { WebvhModuleConfig } from '../WebvhModuleConfig'
+
+import { generateDidDoc } from './didWebvhUtil'
+
+export class WebvhDidRegistrar implements DidRegistrar {
+  public readonly supportedMethods = ['webvh']
+
+  public async create(agentContext: AgentContext, options: WebvhDidCreateOptions): Promise<DidCreateResult> {
+    const didRepository = agentContext.dependencyManager.resolve(DidRepository)
+    const webvhModuleConfig = agentContext.dependencyManager.resolve(WebvhModuleConfig)
+
+    let didDocument: DidDocument
+
+    try {
+      if (options.didDocument) {
+        didDocument = options.didDocument
+      } else if (options.secret?.verificationMethod) {
+        const verificationMethod = options.secret.verificationMethod
+        const privateKeyMultibase = verificationMethod.privateKeyMultibase
+        if (!privateKeyMultibase) {
+          return {
+            didDocumentMetadata: {},
+            didRegistrationMetadata: {},
+            didState: {
+              state: 'failed',
+              reason: 'Invalid private key provided',
+            },
+          }
+        }
+        const privateKey = Buffer.from(MultiBaseEncoder.decode(privateKeyMultibase).data)
+        if (privateKeyMultibase && !isValidPrivateKey(privateKey, KeyType.Ed25519)) {
```

Comment:
Is Ed25519 required? Just curious, we'd like to use it with P256 as well

Reply:
Yes unfortunately ED25519 is required for did:webvh for update keys. We did it that way to standardize it on one interoperable crypto suite you can of course use any keys in a DID document you desire.